src/content/docs/en/reference/image-service-reference.mdx

```diff
@@ -0,0 +1,181 @@
+- **External services** point to URLs and can add support for services such as Cloudinary, Vercel, or any [RIAPI](https://github.com/riapi/riapi)-compliant server.
+
+## Building using the Image Services API
+
+Services definitions take the shape of an exported default object with various required methods ("hooks") to create the required properties for your image. 
+
+Local services have a `transform()` method available to perform transformations on your image. External services can only return the final URL and image attributes.
+
+### Local Service
+
+To create your own local service, you can choose to use the built-in endpoint (`/_image`) and pass your own `transform` function, or you can additionally create your own endpoint that calls the service's functions.
+
+```js
+import type { LocalImageService } from "astro";
+const service: LocalImageService = {
+  getURL(options: ImageTransform) {
+    const href = typeof options.src === "string" ? options.src : options.src.src;
+    return `/my_super_endpoint_that_transforms_images?href={href}&w=${options.width}`;
+  },
+  parseURL(url: URL) {
+    return {
+      src: url.searchParams.get("href"),
+      width: url.searchParams.get("w"),
+    };
+  },
+  transform(buffer: Buffer, options: { src: string, [key: string]: any }): { data: Buffer, format: OutputFormat } {
+    const { buffer } = mySuperLibraryThatEncodesImages(options);
+    return {
+      data: buffer,
+      format: options.format,
+    };
+  },
+};
+export default service;
+```
+
+#### How this Works
+
+At build time for static sites, both `<Image/>` and `getImage(options)` call the `transform()` function. They pass options either through component attributes or an `options` argument, respectively. The transformed images will be built to a `dist/_astro` folder.
+
+In dev mode and SSR mode, Astro doesn't know ahead of time which images need to be optimized. Astro uses a GET endpoint (by default, (`/_image`) to process the images at runtime. `<Image/>` and `getImage()` pass their options to `getURL()`, which will return the endpoint URL. Then, the endpoint calls `parseURL()` and passes the resulting properties to `transform()`.
+
+#### getConfiguredImageService
+
+### External Service
+
+To use an external service . . . 
+
+```js
+import type { ExternalImageService } from "astro";
+
+const service: ExternalImageService = {
+  getURL(options: ImageTransform) {
+    return `https://mysupercdn.com/${options.src}?q=${options.quality}`;
+  },
+};
+
+
+export default service;
+```
+
+## Hooks
+
+### `getURL()`
+
+**Required for local and external services**
+
+`getURL(options: ImageTransform): string`
+
+For local services, this hook returns the URL of the endpoint that generates your image (in SSR and dev mode). The local endpoint that `getURL()` points to must call both `parseURL()` and `transform()`.
```

Comment:
Calling `parseURL` and `transform` is technically optional. You might have cases where your endpoint isn't written in JavaScript, and as such your service won't use `parseURL`